scipy/__init__.py

```diff
@@ -67,7 +67,7 @@
 from scipy._lib import _pep440
 # In maintenance branch, change to np_maxversion N+3 if numpy is at N
 np_minversion = '1.23.5'
-np_maxversion = '9.9.99'
+np_maxversion = '2.5.0'
```

Comment:
We could conditionally bump `np_minversion` to `2.1.3`  if `sysconfig.get_config_var("Py_GIL_DISABLED")` is `1` perhaps? Or maybe something like that--we should have more programmatic freedom to do such a detection here vs. in `pyproject.toml` I'd think.